pkg/transformers/refvars.go

```diff
@@ -0,0 +1,82 @@
+package transformers
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/expansion"
+	"github.com/kubernetes-sigs/kustomize/pkg/resmap"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type refvarTransformer struct {
+	pathConfigs []PathConfig
+	vars        map[string]string
+}
+
+func NewRefVarTransformer(vars map[string]string) (Transformer, error) {
```

Comment:
public method comment.  perhaps

// NewRefVarTransformer returns a Trasformer that replaces $(VAR) style variables with values.

Reply:
This is done.